### PR TITLE
Add fast forward option

### LINKED ISSUE #1010: Implement Option: Skip Frames
I would like to see a option in the debug menu, where you can disable the fps limit, so the game runs faster (like some snes/gba/... emulator can do). It's not about increasing the fps to something higher, so the game runs smoother, but to make it faster. I hope you get the idea :)

- [x] Visual indication
- [x] Android-support

### PR DESCRIPTION
**Pull request by Ghabry**
This adds speedup (x3) via F and x10 via F+PLUS.

Fix #1010

### CODE REVIEW

**Comment by Ghabry** [comment]
What I can observe: The "⏩" is not rendered on my 4.0.3 system...
The speedup is bit buggy, at the beginning the framerate increases (this isn't supposed to happen, must stay at 60) and when you release the key while this happens the game still runs too fast and doesn't return to 60.

**Comment by Ghabry** [comment]
Looks like with 4.1 it shows something, so I only need a fallback ">" for lower apis.

**Comment by Ghabry** [comment]
Theres an unexpected problem here: When Vsync is enabled the fast forward doesn't work because the SDL UpdateDisplay calls block until the VSync happens. To solve this I could spawn a std::thread and do the SDL call there.

**Comment by Ghabry** [comment]
Okay we have a problem here: Video output doesn't work in Android when I do the rendering from an extra thread because the EGL context is on the wrong thread :(

**Comment by BlisterB** [comment]
Arf, android...
If you are afraid that the character ">>" doesn't show on every devices, you can create a class inheriting from the button class, and override the method that draw the button (by drawing it manually). Or you can just add a condition on the drawing method from the button class.

**Comment by Ghabry** [comment]
I already fixed this, by rendering ">" in that case. GOod enough for old devices. The Android stuff won't be part of this PR anymore, because I have to fix it first >.>

File: src/graphics.cpp [hunk 1]
```diff
@@ -189,25 +191,11 @@ void Graphics::DrawFrame() {
 		drawable->Draw();
 	}
 
-	DrawOverlay();
+	//DrawOverlay();
```
Comment (carstene1ns) [inline src/graphics.cpp]: Debug leftover

File: src/input_buttons.h [hunk 2]
```diff
@@ -60,7 +60,7 @@ namespace Input {
 		PAGE_DOWN,
 		SCROLL_UP,
 		SCROLL_DOWN,
-		BUTTON_COUNT
+		FAST_FORWARD,		BUTTON_COUNT
```
Comment (carstene1ns) [inline src/input_buttons.h]: missing `\n`

File: src/player.h [hunk 3]
```diff
@@ -238,6 +243,13 @@ namespace Player {
 	/** Game title. */
 	extern std::string game_title;
 
+	/**
+	 * The default speed modifier applied when the speed up button is pressed
+	 *  Only used for configuring the speedup, don't read this var directly use
+	 *  GetSpeedModifier() instead.
+	 */
+	extern int speed_modifier;
```
Comment (carstene1ns) [inline src/player.h]: Why is this variable needed anyway?

Comment (Ghabry) [inline src/player.h]: The idea was to have an exported variable that a potential setting dialog could alter to get more/less speedup. But currently its useless :D

Comment (carstene1ns) [inline src/player.h]: okay, granted.

**Comment by Ghabry** [comment]
Just found a much smarter way to handle the updating.
Has two disadvantages compared to the old code but on the other hand keeps Player::update almost untouched (before Update contained tons of code):

Input is handled when speedup is on as-if the key was held for 3 (10) frames. Makes navigating while speedup is on a bit tricky (menus are unusable), just turn off the speedup before using the cursor ;).
All audio played in the 3 (10) frames starts simultaniously. Probably not noticable because the difference is a few milliseconds.

**Comment by carstene1ns** [comment]
I get the `Framerate is 0 FPS` message when the gamebrowser launches, seems like the fps counter is not initialized yet or something.
Also, currently sounds are played at full length, this leads to `No free channels available` warnings more often.

**Pull request merged**